### ISSUE TITLE
Sensible and latent heat rejected by AC-units: specifying the state and mass flow of outlet air

### ISSUE THREAD

**Issue by DavidKayanan** [architecture-building-systems/CityEnergyAnalyst#2663]
**Specifying the state and mass flow of outlet air**
As a researcher, I would to calculate the state and mass flow of the output air of cooling towers of air-conditioning systems. 

**Purpose**
This contains the information of the split between latent and sensible heat, and could be used in CFD models (e.g. in AnsysFluent)

**End-User**
Researcher

**Method**
I believe the method can be outlined as follows:
1. setup the energy balance of the cooling tower 
2. fix the states:
- inlet air must be the given ambient conditions
- assume flow rates of water
- calculate the states of the water streams from the total heat rejected by the AC system)
- calculate the state of the outlet air (e.g. T and relative humidity); can either use analytical equations for the state of air, or use a psychrometric chart

3. calculate the air mass flow from the energy balance

**Comment by JIMENOFONSECA** [comment]
@shanshanhsieh mind describing how we do it now in CEA and pointing David to the scripts which may get some of this information?

**Comment by JIMENOFONSECA** [comment]
@DavidKayanan I just checked and we have a very simplified model that only calculates the electrical load. https://github.com/architecture-building-systems/CityEnergyAnalyst/blob/master/cea/technologies/cooling_tower.py

maybe @shanshanhsieh has more information about this?

**Comment by shanshanhsieh** [comment]
Hello @DavidKayanan!
I've sent you some information on the mass balance of cooling towers to your email address.
Please take a look and we can discuss from there!

**Comment by shanshanhsieh** [comment]
Hello @DavidKayanan ,
Could you please describe all of the inputs and outputs of this model? (e.g. inputs: cooling load, outputs: electricity consumption) 
This information will help me to decide how to integrate your model into the CEA.

**Comment by daren-thomas** [comment]
@DavidKayanan: We're thinking of creating a plugin-system for scripts that are not part of the CEA core workflow and would like to make this the first such plugin. I'll reach out to you next week for this.

**Comment by DavidKayanan** [comment]
@daren-thomas This sounds good to me. From the engineering perspective, this step really occurs at the end of the analysis, and is more like post-processing.

**Comment by daren-thomas** [comment]
@DavidKayanan, Issue #2716 deals with a plugin architecture for adding functionality to the CEA and we expect to have this for Release v3.3. I would love to use this issue here as a first proof-of-concept.

**Comment by daren-thomas** [comment]
Status update: We're moving this back to "Product Backlog" as we first need to solve the Python 3 dependency and the plugin architecture. Note: This issue should be completed before end of June.

**Comment by shanshanhsieh** [comment]
Hi @DavidKayanan,

We have finished the migration to python3, so you can add you script as a plug-in now!
Please follow the steps in our blog, and let us know if you have any problem.
https://cityenergyanalyst.com/blog-1

thanks!

**Comment by daren-thomas** [comment]
@jimenofonseca are you interested in making this plugin? @DavidKayanan seems to have moved on.

**Comment by jimenofonseca** [comment]
Not immidiately.
Best, Jimeno

**Comment by jimenofonseca** [comment]
Adding here some info from @DavidKayanan in regards to this topic

[CT Performance Curves v1.1.pdf](https://github.com/architecture-building-systems/CityEnergyAnalyst/files/5651003/CT.Performance.Curves.v1.1.pdf)
[CEA tech - cooling towers.docx](https://github.com/architecture-building-systems/CityEnergyAnalyst/files/5651005/CEA.tech.-.cooling.towers.docx)

**Comment by shanshanhsieh** [comment]
there is no plan to build a plugin at the moment, feel free re-open this issue if there are further interests in using this script.
the script is residing in this open repository: https://github.com/cooling-singapore/CoolingTower

**Comment by lguilhermers** [comment]
A plugin is on the work, residing here: https://github.com/cooling-singapore/cea-heat-rejection-plugin

**Comment by lguilhermers** [comment]
> A plugin is on the work, residing here: https://github.com/cooling-singapore/cea-heat-rejection-plugin

A working version is already available on the mentioned link. @shanshanhsieh maybe you want to add it in the list of plugins in the website.

**Comment by shanshanhsieh** [comment]
done!
